### PR TITLE
Add basic 11ty dev environment.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
- Move old files to `/old_source/` dir
- Add basic [11ty](https://www.11ty.dev/) config, with [Liquid](https://liquidjs.com/) templates
- Add Sass and JS (via Rollup) compilation
- Add TypeScript, Prettier, ESLint, and Stylelint
- Add CI and upgrade-deps workflows (and remove failing old CI workflow)

The site templates have not been ported over yet -- this is just an 11ty dev env, with a sample base layout and footer include.

Todo:

- [x] Update readme/contributing docs

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[11ty/Typescript dev environment](https://trello.com/c/gBwlltqS/1-11ty-typescript-dev-environment)

File: .stylelintrc.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  color-named: always-where-possible
+  custom-property-pattern: null
+  function-url-no-scheme-relative: true
+  number-max-precision: null
+  selector-class-pattern: null
+
+  # Sass
+  scss/at-mixin-pattern: null
+  scss/dollar-variable-pattern: null
+  scss/no-global-function-names: null
```
Comment (jgerigmeyer) [inline .stylelintrc.yml:30]: I added these to quiet errors, but we probably want to enable and fix them instead? And review/edit the other Stylelint rules...

**Comment by jgerigmeyer** [comment]
This is ready for initial review -- probably at least from @mirisuzanne and @stacyk, then @sanajaved7 and/or @jerivas or others if you like?

**Review by jerivas — APPROVED** [review]
I'm able to run the 11ty site on localhost:8080, LGTM. I also added container configuration files if someone is interested in running that way

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM node:16
+
+WORKDIR /app
+
+COPY . /app
+RUN yarn install
```
Comment (jgerigmeyer) [inline Dockerfile:6]: @jerivas Is there a reason you didn't add any `CMD` here?

**Comment by jgerigmeyer** [comment]
[Migrate to Typescript (JS)](https://trello.com/c/zO4MFhK4/7-migrate-to-typescript-js)

**Comment by jgerigmeyer** [comment]
[Basic Migration to Dart Sass](https://trello.com/c/lnlCYcnL/4-basic-migration-to-dart-sass)

File: Dockerfile (hunk 2 above)
Comment (jerivas) [reply Dockerfile:6]: Oh, good catch. I think we want `yarn serve` in the compose file for development, but what should be the command for production? `yarn build`?

Comment (jgerigmeyer) [reply Dockerfile:6]: At this point I only envision Docker being used for development, so I might start with `yarn serve` in both cases? But `yarn build-prod` is fine too.

**Review by mirisuzanne — COMMENTED** [review]
This looks good as an initial setup. The Liquid templates will take some getting used to. Not sure if we're planning to do the Sass migration steps in this branch/PR? I didn't review any of the styles in detail since they haven't really changed - only moved.

File: Dockerfile (hunk 2 above)
Comment (jerivas) [reply Dockerfile:6]: Add `yarn serve` in 43b979cfd3c865c64ed9211702fed45022a9c605

**Comment by jgerigmeyer** [comment]
> Not sure if we're planning to do the Sass migration steps in this branch/PR?

Yeah, I think that cleanup can happen in another PR.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,47 +1,74 @@
...
+- Write a detailed description of what you're adding in the pull request
   (screenshots help).
-* If there is new design or CSS, please add @Jina as a reviewer so she can see
+- If there is new design or CSS, please add @Jina as a reviewer so she can see
   if it needs to be added to the style guide (or if a suitable alternative
   exists).
-* Submit the pull request to the `main` branch.
-* Drink whisky.
+- Submit the pull request to the `main` branch.
+- Drink whisky.
```
Comment (stacyk) [inline CONTRIBUTING.md:14]: 🍹

Comment (stacyk) [reply CONTRIBUTING.md:14]: TIL "Whisky (no e) refers to Scottish, Canadian, or Japanese grain spirits. Whiskey (with an e) refers to grain spirits distilled in Ireland and the United States."

**Comment by jgerigmeyer** [comment]
[Data Files & 11ty config for site architecture](https://trello.com/c/Xixp2Inp/2-data-files-11ty-config-for-site-architecture)

**Review by stacyk — APPROVED** [review]
Looks good the second time I ran the commands :)

**Pull request merged**